### PR TITLE
Add manual testing steps in DG

### PR DESCRIPTION
**Pull request by dandaandaaaaaan**
This PR covers the manual testing steps for the DG for my parts, plus some slight housekeeping stuff

Live version [here](https://dandaandaaaaaan.github.io/tp/DeveloperGuide.html)

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1237,38 +1237,59 @@ testers are expected to do more *exploratory* testing.
...
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
-
-1. _{ more test cases …​ }_
+   1. Test case: Missing data file<br>
+      Expected: Default contacts are loaded instead. Upon an operation that interacts with the data, e.g. add/delete/edit persons, the data will then be saved as `addressbook.json`.
+   
+   2. Test case: Corrupted JSON Formatting in data file<br>
```
Comment (sharpstorm) [inline docs/DeveloperGuide.md:1256]: I suggest changing the wording to make it clearer.
```suggestion
   2. Test case: Corrupted JSON Structure in data file<br>
```

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -1237,38 +1237,59 @@ testers are expected to do more *exploratory* testing.
...
+      Expected: An empty AddressBook will be loaded instead. Upon an operation that interacts with the data, the existing corrupted `addressbook.json` file will be replaced with the new data.
+   
+   3. Test case: Invalid entry in individual Person data in data file (e.g. alphabets in phone number)<br>
```
Comment (sharpstorm) [inline docs/DeveloperGuide.md:1259]: I suggest making it clearer that this is JSON data file.
```suggestion
   3. Test case: Invalid Person record in JSON data file (e.g. alphabets in phone number)<br>
```

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -1237,38 +1237,59 @@ testers are expected to do more *exploratory* testing.
...
-
 ### Saving data
 
 1. Dealing with missing/corrupted data files
 
-   1. _{explain how to simulate a missing/corrupted file, and the expected behavior}_
-
-1. _{ more test cases …​ }_
+   1. Test case: Missing data file<br>
+      Expected: Default contacts are loaded instead. Upon an operation that interacts with the data, e.g. add/delete/edit persons, the data will then be saved as `addressbook.json`.
```
Comment (sharpstorm) [inline docs/DeveloperGuide.md:1254]: Can you bold the expected for all of the cases?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -1237,38 +1237,59 @@ testers are expected to do more *exploratory* testing.
...
+   3. Test case: Editing the entries but with invalid data <br>
+      Expected: Upon opening of ContaX, the particular entry will be skipped.
+
+### Importing Contacts from CSV Files
+
+1. Importing contacts from a CSV File
+
+   1. Prerequisites: Must have an existing CSV File with contact information.
+   
+   2. Test case: CSV File was exported previously from another instance of ContaX, or is in proper ContaX format <br>
```
Comment (sharpstorm) [inline docs/DeveloperGuide.md:1278]: I suggest linking to the CSV format section.

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (dandaandaaaaaan) [reply docs/DeveloperGuide.md]: thanks for catch, fixed in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/341/commits/c96996c17298418ba564eb619e4b83c4cb7e62c3

**Review by sharpstorm — APPROVED** [review]
LGTM.

File: docs/DeveloperGuide.md (hunk 2 above)
Comment (dandaandaaaaaan) [reply docs/DeveloperGuide.md]: Thanks for catch, added in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/341/commits/c96996c17298418ba564eb619e4b83c4cb7e62c3

File: docs/DeveloperGuide.md (hunk 3 above)
Comment (dandaandaaaaaan) [reply docs/DeveloperGuide.md]: done in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/341/commits/c96996c17298418ba564eb619e4b83c4cb7e62c3

File: docs/DeveloperGuide.md (hunk 4 above)
Comment (dandaandaaaaaan) [reply docs/DeveloperGuide.md]: done in https://github.com/AY2122S2-CS2103-W17-1/tp/pull/341/commits/c96996c17298418ba564eb619e4b83c4cb7e62c3

**Review by Wei-Xinn — APPROVED** [review]
lgtm

**Pull request merged**